src/values/AbstractObjectValue.js

```diff
@@ -677,6 +677,34 @@ export default class AbstractObjectValue extends AbstractValue {
         return cv.$Set(P, V, Receiver === this ? cv : Receiver);
       }
       invariant(false);
+    } else if (this.kind === "conditional") {
+      // this is the join of two concrete/abstract objects
+      // use this join condition for the join of the two property values
+      let [cond, ob1, ob2] = this.args;
+      invariant(cond instanceof AbstractValue);
+      invariant(ob1 instanceof ObjectValue || ob1 instanceof AbstractObjectValue);
+      invariant(ob2 instanceof ObjectValue || ob2 instanceof AbstractObjectValue);
+      let d1 = ob1.$GetOwnProperty(P);
+      let d2 = ob2.$GetOwnProperty(P);
```

Comment:
Found an existing bug. These don't consider setters on the prototype chain. https://github.com/facebook/prepack/issues/2226